api/src/middleware/authenticate.ts

```diff
@@ -11,17 +11,33 @@ import isJWT from '../utils/is-jwt';
  */
 const authenticate: RequestHandler = asyncHandler(async (req, res, next) => {
 	req.accountability = {
+		maintenance: false,
+		maintenance_role: null,
 		user: null,
+		user_role: null,
 		role: null,
 		admin: false,
 		app: false,
 		ip: req.ip.startsWith('::ffff:') ? req.ip.substring(7) : req.ip,
 		userAgent: req.get('user-agent'),
 	};
 
-	if (!req.token) return next();
-
 	const database = getDatabase();
+	const settings = await database
+		.select('directus_settings.maintenance_active', 'directus_settings.maintenance_role')
+		.from('directus_settings')
+		.first();
+
+	if (settings && settings.maintenance_active) {
+		req.accountability.maintenance = true;
+		req.accountability.maintenance_role = settings.maintenance_role;
+
+		if (!settings.maintenance_role) {
```

Comment:
We have to be careful around these values straight from the DB. Some databases might return `0` instead of `false`, some might return `"FALSE"`, or `"0"` which can cause trouble in these falsy checks